src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,271 @@
+
+    private static int numberOfBarbarians = 0;
+    private static int numberOfArchers    = 0;
+    private static int numberOfChiefs     = 0;
+
+    private static final int MAX_SIZE = Configurations.maxBarbarianHordeSize;
+
+    private static final double BARBARIANS_MULTIPLIER        = 0.5;
+    private static final double ARCHER_BARBARIANS_MULTIPLIER = 0.25;
+    private static final double CHIEF_BARBARIANS_MULTIPLIER  = 0.1;
+
+    private static final int PREFERRED_MAX_HORDE_SIZE = 40;
+    private static final int PREFERRED_MAX_BARBARIANS = 22;
+    private static final int PREFERRED_MAX_ARCHERS    = 16;
+    private static final int PREFERRED_MAX_CHIEFS     = 2;
+
+    private static final float WHOLE_CIRCLE  = 360.0F;
+    private static final float HALF_A_CIRCLE = 180F;
+
+    /**
+     * Private constructor to hide the implicit public one.
+     */
+    private BarbarianUtils()
+    {
+    }
+
+    /**
+     * Takes a colony and spits out that colony's RaidLevel.
+     *
+     * @param colony The colony to use
+     * @return an int describing the raid level
+     */
+    public static int getColonyRaidLevel(final Colony colony)
+    {
+        int levels = 0;
+
+        @NotNull final List<CitizenData> citizensList = new ArrayList<>();
+        citizensList.addAll(colony.getCitizens().values());
+
+        for (@NotNull final CitizenData citizen : citizensList)
+        {
+            if (citizen.getJob() != null && citizen.getWorkBuilding() != null)
+            {
+                final int buildingLevel = citizen.getWorkBuilding().getBuildingLevel();
+                levels += buildingLevel;
+            }
+        }
+
+        if (colony.getTownHall() != null)
+        {
+            return (levels + colony.getTownHall().getBuildingLevel());
+        }
+        else
+        {
+            return 0;
+        }
+    }
+
+    /**
+     * Returns the closest barbarian to an entity.
+     *
+     * @param entity             The entity to test against
+     * @param distanceFromEntity The distance to check for
+     * @return the barbarian (if any) that is nearest
+     */
+    public static AbstractEntityBarbarian getClosestBarbarianToEntity(final Entity entity, final double distanceFromEntity)
+    {
+        Optional<AbstractEntityBarbarian> barbarian = getBarbariansCloseToEntity(entity, distanceFromEntity).stream().findFirst();
+        return barbarian.orElse(null);
+    }
+
+    /**
+     * Returns the barbarians close to an entity.
+     *
+     * @param entity             The entity to test against
+     * @param distanceFromEntity The distance to check for
+     * @return the barbarians (if any) that is nearest
+     */
+    public static List<AbstractEntityBarbarian> getBarbariansCloseToEntity(final Entity entity, final double distanceFromEntity)
+    {
+        final List<AbstractEntityBarbarian> entityList = CompatibilityUtils.getWorld(entity).getEntitiesWithinAABB(
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "entityList". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1488)